File: scripts/extract-flow.js
import { createReadStream } from 'fs';
import { mkdir, writeFile, readdir, readFile } from 'fs/promises';
import { createRequire } from 'module';
import { execSync } from 'child_process';

// Extract the zip file
try {
  execSync('unzip -o /vercel/share/v0-project/Dosage_20260313041844.zip -d /vercel/share/v0-project/flow-extracted', { stdio: 'inherit' });
  console.log('Extraction complete!');
  
  // List extracted files
  const files = execSync('find /vercel/share/v0-project/flow-extracted -type f', { encoding: 'utf-8' });
  console.log('\nExtracted files:');
  console.log(files);
} catch (error) {
  console.error('Error:', error.message);
}

File: scripts/extract_flow.py
import zipfile
import os
import json

zip_path = '/vercel/share/v0-project/Dosage_20260313041844.zip'
output_dir = '/vercel/share/v0-project/flow-extracted'

# Create output directory
os.makedirs(output_dir, exist_ok=True)

# Extract the zip file
with zipfile.ZipFile(zip_path, 'r') as zip_ref:
    print("Files in zip:")
    for name in zip_ref.namelist():
        print(f"  - {name}")
    
    zip_ref.extractall(output_dir)
    print(f"\nExtracted to: {output_dir}")

# List extracted files
print("\nExtracted files:")
for root, dirs, files in os.walk(output_dir):
    for file in files:
        filepath = os.path.join(root, file)
        print(f"  - {filepath}")
        
        # If it's a JSON file, try to read and display structure
        if file.endswith('.json'):
            try:
                with open(filepath, 'r', encoding='utf-8') as f:
                    data = json.load(f)
                    print(f"    Keys: {list(data.keys()) if isinstance(data, dict) else 'Array'}")
            except Exception as e:
                print(f"    Could not parse: {e}")
